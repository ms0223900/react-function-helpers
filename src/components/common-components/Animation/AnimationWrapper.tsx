import { Box } from '@material-ui/core';
import { HOC } from 'all-common-types';
import React, { ReactElement, ReactNode } from 'react';
import 'styles/styles.scss';

const defaultAnimate = 'slideInDown';
const defaultAnimationDuration = 1;
const defaultAnimationDelay = 0;

export interface AnimationWrapperWithoutChildrenProps {
  animateEffect?: string
  animationDelay?: number
  animationDuration?: number
}
export interface AnimationWrapperProps extends AnimationWrapperWithoutChildrenProps {
  children: ReactNode
}

const AnimationWrapper = ({
  animateEffect=defaultAnimate,
  animationDelay=defaultAnimationDelay,
  animationDuration=defaultAnimationDuration,
  children,
}: AnimationWrapperProps) => {
  return (
    <Box 
      style={{ 
        animationDuration: `${animationDuration - animationDelay}s`,
        animationDelay: `${animationDelay}s`,
        // animationIterationCount: 'infinite',
        animationFillMode: 'forwards', 
      }}  
      className={`${animateEffect}`}
    >
      {children}
    </Box>
  );
};

type AnimationHOC = HOC<AnimationWrapperWithoutChildrenProps>
export const AnimationHOC: AnimationHOC = (Component) => {
  return props => (
    <AnimationWrapper {...props}>
      <Component {...props} />
    </AnimationWrapper>
  );
};

export default AnimationWrapper;