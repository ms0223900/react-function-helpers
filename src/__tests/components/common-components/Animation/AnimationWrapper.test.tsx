import React from 'react';
import { shallow, mount } from 'enzyme';
import renderer from 'react-test-renderer';
import AnimationWrapper, { AnimationHOC } from 'components/common-components/Animation/AnimationWrapper';

describe('test AnimationWrapper', () => {
  it('should render correctly(with default props)', () => {
    const tree = renderer.create(
      <AnimationWrapper>
        <></>
      </AnimationWrapper>
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should render correctly(with custom props)', () => {
    const tree = renderer.create(
      <AnimationWrapper 
        animateEffect={'slideOutDown'}
        animationDuration={10}
      >
        <></>
      </AnimationWrapper>
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('HOC should render correctly', () => {
    const MockComponent = (props: { test: string }) => <></>;
    const AnimatedComponent = AnimationHOC(MockComponent);
    const tree = renderer.create(
      <AnimatedComponent test={''} />
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});