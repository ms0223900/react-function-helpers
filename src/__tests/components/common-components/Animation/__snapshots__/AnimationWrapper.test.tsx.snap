// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test AnimationWrapper HOC should render correctly 1`] = `
<div
  className="MuiBox-root MuiBox-root-4 slideInDown"
  style={
    Object {
      "animationDelay": "0s",
      "animationDuration": "1s",
      "animationFillMode": "forwards",
    }
  }
/>
`;

exports[`test AnimationWrapper should render correctly(with custom props) 1`] = `
<div
  className="MuiBox-root MuiBox-root-3 slideOutDown"
  style={
    Object {
      "animationDelay": "0s",
      "animationDuration": "10s",
      "animationFillMode": "forwards",
    }
  }
/>
`;

exports[`test AnimationWrapper should render correctly(with default props) 1`] = `
<div
  className="MuiBox-root MuiBox-root-2 slideInDown"
  style={
    Object {
      "animationDelay": "0s",
      "animationDuration": "1s",
      "animationFillMode": "forwards",
    }
  }
/>
`;
